location-service/src/main/java/com/telecominfraproject/wlan/location/service/LocationServiceController.java

```diff
@@ -93,9 +94,17 @@ public Location update(@RequestBody Location location) {
             LOG.error("Failed to update location, request contains unsupported value: {}", location);
             throw new DsDataValidationException("Location contains unsupported value");
         }
+        
+        Location existingLocation = locationDatastore.get(location.getId());
 
         Location ret = locationDatastore.update(location);
         
+        if (existingLocation != null && existingLocation.getDetails() != null && ret != null && ret.getDetails() != null &&
+        		!existingLocation.getDetails().getCountryCode().equals(ret.getDetails().getCountryCode())) {
+        	LocationChangedApImpactingEvent apImpactingEvent = new LocationChangedApImpactingEvent(ret);
+            publishEvent(apImpactingEvent);
```

Comment:
when more the one event is to be published - please use the publishEvents or publishEventsBulk methods of the CloudEventDispatcherInterface